jax/linear_util.py

```diff
@@ -12,6 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+"""
+Utilities for defining linear functions composed with transformations.
+
+"Linear" here is meant in the sense of linear types; that is, a linear function
+may be called at most once.
+
+For example:
+
+from jax import linear_util as lu
+
+# A transformation that scales its argument down and its result up.
+@lu.transformation
+def scale_transformer(scale, x):
+  ans = yield (x / scale,)
+  yield (x * scale,)
```

Comment:
This doesn't have to be a tuple actually. It's for the usual reason of functions having multiple arguments but single outputs.

Reply:
Done.